rules-tests/Php74/Rector/Property/TypedPropertyRector/FixtureUnionIntersectionTypes/multiple_types.php.inc

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Tests\Php74\Rector\Property\TypedPropertyRector\FixtureUnionIntersectionTypes;
+
+final class MultipleTypes
+{
+    /**
+     * @var string|int|float
+     */
+    private $nonNullableVariable;
+    
+    /**
+     * @var string|int|float|null
+     */
+    private $nullableVariable;
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php74\Rector\Property\TypedPropertyRector\FixtureUnionIntersectionTypes;
+
+final class MultipleTypes
+{
+    private string|int|float $nonNullableVariable;
+    
+    private string|int|float|null $nullableVariable = null;
+}
+
+?>
```

Comment:
Ok so we have two issues:

1. `UnionTypesRector` for PHP 8.0 relies on `TypedPropertiesRector` for properties which is only for PHP 7.4
2. `TypedPropertiesRector` updates to union type even if not supported by 7.4 if there is no nullable type (see: https://getrector.org/demo/1ec8a983-982c-6f22-b892-3f6226af2d4b)

Reply:
Actually, it is because of feature set not configured in php 7.4, so in php 7.4, it already valid, ref

https://getrector.org/demo/1ec8a9b2-9990-6bdc-8e11-91fdaf426290

You just need to configure `Option::PHP_VERSION_FEATURES`:

```php
use Rector\Core\Configuration\Option;

use Rector\Php74\Rector\Property\TypedPropertyRector;
use Rector\Set\ValueObject\SetList;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
use Rector\Core\ValueObject\PhpVersionFeature;

return static function (ContainerConfigurator $containerConfigurator): void {
    $services = $containerConfigurator->services();
    $services->set(TypedPropertyRector::class);
    
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::PHP_VERSION_FEATURES, PhpVersionFeature::UNION_TYPES - 1);
};
```